scipy/optimize/tests/test_linprog.py

```diff
@@ -687,6 +687,75 @@ def cb(xk, **kwargs):
         assert_(callback_complete[0])
         assert_allclose(last_xk[0], res.x)
 
+    def test_bug_8662(self):
+        # scipy.linprog returns incorrect optimal result for constraints using
+        # default bounds, but, correct if boundary condition as constraint.
+        # https://github.com/scipy/scipy/issues/8662
+        c = [-10, 10, 6, 3]
+        A = [
+            [8, -8, -4, 6],
+            [-8, 8, 4, -6],
+            [-4, 4, 8, -4],
+            [3, -3, -3, -10]
+        ]
+        b = [9, -9, -9, -4]
+        bounds = [(0, None), (0, None), (0, None), (0, None)]
+
+        res1 = linprog(c, A, b, bounds=bounds)
+        _assert_success(res1, 36.0000000000)
+
+        # Set boundary condition as a constraint
+        A.append([0, 0, -1, 0])
+        b.append(0)
+        bounds[2] = (None, None)
+
+        res2 = linprog(c, A, b, bounds=bounds)
+        _assert_success(res2, 36.0000000000)
+        assert_allclose(res1.x, res2.x)
+
+    def test_bug_6690_with_simplex(self):
```

Comment:
As #6690 is tested for `method='interior-point'` already, rather than adding this here, consider moving `test_bug_6690` out of `BaseTestLinprogIP` into `LinprogCommonTests`.  However this might mess up the `@pytest.mark.xfail` for this test in `TestLinprogIPSparse` and `TestLinprogIPSparsePresolve` unless those lines 

`super(TestLinprogIPSparse, self).test_bug_6690()` and
`super(TestLinprogIPSparsePresolve, self).test_bug_6690()`

are modified. I am not sure what is recommended here - help @rgommers? Can they both just be:

`LinprogCommonTests.test_bug_6690(self)`?